openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java

```diff
@@ -146,18 +149,28 @@ private static boolean containErrors(List<Diagnostic> diagnostics) {
     private static void extractListenersAndServiceNodes(String serviceName, List<String> availableService,
                                                         List<ServiceDeclarationNode> servicesToGenerate,
                                                         ModulePartNode modulePartNode,
-                                                        List<ListenerDeclarationNode> endpoints) {
+                                                        List<ListenerDeclarationNode> endpoints,
+                                                        SemanticModel semanticModel) {
         for (Node node : modulePartNode.members()) {
             SyntaxKind syntaxKind = node.kind();
             // Load a listen_declaration for the server part in the yaml spec
             if (syntaxKind.equals(SyntaxKind.LISTENER_DECLARATION)) {
                 ListenerDeclarationNode listener = (ListenerDeclarationNode) node;
                 endpoints.add(listener);
             }
-            // Load a service Node
             if (syntaxKind.equals(SyntaxKind.SERVICE_DECLARATION)) {
-                ServiceDeclarationNode serviceNode = (ServiceDeclarationNode) node;
-                extractServiceDeclarationNodes(serviceName, availableService, servicesToGenerate, serviceNode);
+                /**
+                 * Here check the service is related to the http module by checking listener type that attached to
+                 * service endpoints.
+                 */
+                ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) node;
+                Optional<Symbol> serviceSymbol = semanticModel.symbol(serviceDeclarationNode);
+                if (serviceSymbol.isPresent() && serviceSymbol.get() instanceof ServiceDeclarationSymbol) {
+                    List<TypeSymbol> listenerTypes = ((ServiceDeclarationSymbol) serviceSymbol.get()).listenerTypes();
```

Comment:
IMO we better move all these logic into the util function, so that we can simply validate services just by passing the service node. WDYT?

Reply:
Here `isHttpListener` function is validated by looking at the listener type of service node whether it is http service or not, hereafter we have to pass this ` serviceName, availableService, servicesToGenerate ` arguments to fill the available services in case the user has added the wrong service name to command to generate spec and return the error message with available services.  `servicesToGenerate` need to store every possible service nodes in one ballerina file to generate spec when a user hasn't mentioned specific service to generate. 
Hope we can simplify this with upcoming improvements, Shall I try to simplify this with improvements 🙂

Reply:
sure 🙂